pkg/plugins/policies/core/rules/resourcerules.go

```diff
@@ -0,0 +1,257 @@
+
+	var resolvedItems []*resolvedPolicyItem
+	for _, item := range list {
+		resolvedItems = append(resolvedItems, resolveTargetRef(item, reader)...)
+	}
+
+	// we could've built ResourceRule for all resources in the cluster, but we only need to build rules for resources
+	// that are part of the policy to reduce the size of the ResourceRules
+	for uri, resource := range indexResources(resolvedItems) {
+		// take only policy items that have isRelevant conf for the resource
+		var relevant []*resolvedPolicyItem
+		for _, policyItem := range resolvedItems {
+			if isRelevant(policyItem, resource, uri.SectionName) {
+				relevant = append(relevant, policyItem)
+			}
+		}
+
+		if len(relevant) > 0 {
+			// merge all relevant confs into one, the order of merging is guaranteed by SortByTargetRefV2
+			merged, err := mergeConfs(relevant)
+			if err != nil {
+				return nil, err
+			}
+			rules[UniqueResourceKey(uri.String())] = ResourceRule{
+				Resource:            resource.GetMeta(),
+				ResourceSectionName: uri.SectionName,
+				Conf:                merged,
+				Origin:              origins(relevant),
+			}
+		}
+	}
+
+	return rules, nil
+}
+
+func indexResources(ri []*resolvedPolicyItem) map[UniqueResourceIdentifier]core_model.Resource {
+	index := map[UniqueResourceIdentifier]core_model.Resource{}
+	for _, i := range ri {
+		index[uniqueKey(i.resource, i.sectionName())] = i.resource
+	}
+	return index
+}
+
+func mergeConfs(ri []*resolvedPolicyItem) ([]interface{}, error) {
+	var confs []interface{}
+	for _, i := range ri {
+		confs = append(confs, i.item.GetDefault())
+	}
+	return MergeConfs(confs)
+}
+
+func origins(ri []*resolvedPolicyItem) []Origin {
+	var rv []Origin
+
+	type keyType struct {
+		core_model.ResourceKey
+		ruleIndex int
+	}
+	key := func(policyItem PolicyItemWithMeta) keyType {
+		return keyType{
+			ResourceKey: core_model.MetaToResourceKey(policyItem.ResourceMeta),
+			ruleIndex:   policyItem.RuleIndex,
+		}
+	}
+	set := map[keyType]struct{}{}
+	for _, i := range ri {
+		if _, ok := set[key(i.item)]; !ok {
+			rv = append(rv, Origin{Resource: i.item.ResourceMeta, RuleIndex: i.item.RuleIndex})
+			set[key(i.item)] = struct{}{}
+		}
+	}
+	return rv
+}
+
+func uniqueKey(r core_model.Resource, sectionName string) UniqueResourceIdentifier {
+	return UniqueResourceIdentifier{
+		ResourceIdentifier: core_model.NewResourceIdentifier(r),
+		ResourceType:       r.Descriptor().Name,
+		SectionName:        sectionName,
+	}
+}
+
+// isRelevant returns true if the policyItem is relevant to the resource or section of the resource
+func isRelevant(policyItem *resolvedPolicyItem, r core_model.Resource, sectionName string) bool {
+	switch policyItem.resource.Descriptor().Name {
+	case core_mesh.MeshType:
+		switch r.Descriptor().Name {
+		case core_mesh.MeshType:
+			return policyItem.resource.GetMeta().GetName() == r.GetMeta().GetName()
+		default:
+			return policyItem.resource.GetMeta().GetName() == r.GetMeta().GetMesh()
```

Comment:
`policyItem.resource.GetMeta().GetName() == r.GetMeta().GetMesh()` should we even check this? wouldn't it end up in merge all other types with Mesh? I think that we've discussed that we won't merge MeshService conf with Mesh conf at this level?